dev/doc/README.md

```diff
@@ -6,25 +6,30 @@ See the first section of [`INSTALL`](../../INSTALL).  Developers are
 recommended to use a recent OCaml version, which they can get through
 opam or Nix, in particular.
 
-## Building `coqtop`
-The general workflow is to first setup a development environment with
-the correct `configure` settings, then hacking on Coq, make-ing, and testing.
+## Building `coqtop` / `coqc` binaries
 
+The recommended development workflow is to rely on Dune.
+See [`build-system.dune.md`](build-system.dune.md).
 
-This document will use `$JOBS` to refer to the number of parallel jobs one
-is willing to have with `make`.
+```
+$ git clone https://github.com/coq/coq.git
+$ cd coq
+$ make -f Makefile.dune # to get an idea of the available targets
+$ make -f Makefile.dune check # an example of superfast target
+$ dune exec -- dev/shim/coqc-prelude test.v # to run coqc on a test file
+```
 
+Alternatively, one can use the legacy build system (which now relies
+on Dune for the OCaml parts):
 
 ```
-$ git clone git clone https://github.com/coq/coq.git
-$ cd coq
 $ ./configure -profile devel
 $ make -j $JOBS # Make once for `merlin`(autocompletion tool)
 
 <hack>
 
 $ make -j $JOBS states # builds just enough to run coqtop
-$ bin/coqtop -compile <test_file_name.v>
+$ COQLIB=_build_vo/default/ _build/install/default/bin/coqc <test_file_name.v>
```

Comment:
Actually setting this variable shouldn't be needed after #14176 , instead we can just call `_build_vo/default/bin/coqc` and that should work.

Reply:
Cool! Do you think you can finish this PR soon?

Reply:
Yes I'm working on it now.